src/main/java/org/cryptomator/common/LazyProcessedProperties.java

```diff
@@ -0,0 +1,73 @@
+package org.cryptomator.common;
+
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.Properties;
+import java.util.regex.Pattern;
+
+public class LazyProcessedProperties extends PropertiesDecorator {
+
+	private static final Pattern TEMPLATE = Pattern.compile("@\\{(\\w+)}");
+
+	private final Map<String, String> env;
+
+	public LazyProcessedProperties(Properties props, Map<String, String> systemEnvironment) {
+		super(props);
+		this.env = systemEnvironment;
+	}
+
+	@Override
+	public String getProperty(String key) {
+		var value = delegate.getProperty(key);
+		if (key.startsWith("cryptomator.") && value != null) {
+			return process(value);
+		} else {
+			return value;
+		}
+	}
+
+	@Override
+	public String getProperty(String key, String defaultValue) {
+		var value = delegate.getProperty(key, defaultValue);
+		if (key.startsWith("cryptomator.") && value != null) {
+			return process(value);
+		} else {
+			return value;
+		}
+	}
+
+	//visible for testing
+	String process(String value) {
+		return TEMPLATE.matcher(value).replaceAll(match -> //
+				switch (match.group(1)) {
+					case "appdir" -> resolveFrom("APPDIR", Source.ENV);
+					case "appdata" -> resolveFrom("APPDATA", Source.ENV);
+					case "localappdata" -> resolveFrom("LOCALAPPDATA", Source.ENV);
+					case "userhome" -> resolveFrom("user.home", Source.PROPS);
```

Comment:
if we expose `user.home` via `Environment.java`, we can remove `Source`, making the code simpler.

Reply:
I assume some misunderstanding happend here: `Source.ENV` points to the OS environment (retrievable by `System.getenv()`), not the Cryptomator internal class